seqr/views/utils/dataset_utils.py

```diff
@@ -125,6 +126,7 @@ def match_sample_ids_to_sample_records(
         loaded_date (object): datetime object
         raise_no_match_error (bool): whether to raise an exception if no sample matches are found
         raise_unmatched_error_template (string): optional template to use to raise an exception if samples are unmatched, will not raise if not provided
+        sample_id_to_tissue_type (object): a mapping from sample ids to tissue types
```

Comment:
I think it would actually be better (and more in keeping with this function) to do one call to this function per tissue type, and explicitly pass in one tissue type here that will be passed to `_find_matching_sample_records` and sample creation the same way as all the other arguments.

Technically this is less performant becuase its one database query per tissue type instead of one total, but it will make the ongoing maintenance of this function and buggyness of htis code much better, and the performance will still be acceptable